backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/objcexport/BlockPointerSupport.kt

```diff
@@ -188,7 +188,7 @@ internal class BlockAdapterToFunctionGenerator(val objCExportCodeGenerator: ObjC
     }
 
     fun org.jetbrains.kotlin.backend.konan.Context.LongInt(value: Long) =
-            if (is64Bit()) Int64(value) else Int32(value.toInt())
+            if (is64BitNSInteger()) Int64(value) else Int32(value.toInt())
```

Comment:
And this is still `long`, not an `NS(U)Integer`.
That's why **both** predicates are actually required, `is64BitNSInteger` and `is64BitLong`.

Reply:
Ok, I'll create a separate PR for that.